### PR TITLE
Support youtube embeds in entries | Fixes #47

### LINKED ISSUE #47: We need to support youtube embeds in entries
# Problem

Sometimes a video expresses a new feature better than words. We should be able to embed a video link to youtube, vimeo etc. and have it display nicely in the changelog. I do not envision providing video upload support at this point, though it is certainly something we could add later.

# Proposed solution

* Add a new field to the entry model for a user video.
* Make a pleasing layout so that the changelog looks good with all combinations of video, picture, neither being attached to an entry.
* MAke sure that non printable outputs (e.g. rst download) still work nicely.

# Expected outcome

* Writers should be able to insert videos to an entry
* Viewers should be able to see videos on a version's changelog

### PR DESCRIPTION
**Pull request by rischanlab**
Hi @timlinux based on issue #47 could you please check this PR. 
This is the screenshoot. Please let me know if there is any something to be added or edited. 

screen shot 2015-10-12 at 2 03 08 pm

thanks

### CODE REVIEW

File: django_project/changes/models/entry.py [hunk 1]
```diff
@@ -61,6 +62,11 @@ class Entry(AuditedModel):
         null=True,
         blank=True)
 
+    video = EmbedVideoField(verbose_name='Youtube video',
```
Comment (timlinux) [inline django_project/changes/models/entry.py]: pull left

File: django_project/changes/templates/entry/detail.html [hunk 2]
```diff
@@ -63,6 +64,11 @@ <h4 class="muted">Version:
         </a>
       {%  endif %}
     </div>
+  <div class="col-lg-8">
+      {% if entry.video %}
```
Comment (timlinux) [inline django_project/changes/templates/entry/detail.html]: Should the if check not go outside the div?

**Comment by timlinux** [comment]
What about migrations @rischanlab - can you add the needed migrations so that we can deploy on existing database?

**Comment by rischanlab** [comment]
@timlinux That is the problem from the first. At the first time I clone this project I can't do migrate. So I confused then you give me the db.dump so I can run it in my machine. Without that db I can't run it because I can't do migrate. I am still looking the reason. I've tried to makemigrations and migrate the applications one by one but still not success. 

I've mentioned in issue #115 . I guess that is why I cant run `coverage` caused the sistem can't create the database for testing. Hope I can solved as soon as possible. 

thanks for your comment Tim :+1:

**Comment by rischanlab** [comment]
Btw @timlinux for creating superuser (manage createsuperuser), we can use fake initial migrate like this `python manage.py migrate --fake-initial` and it works

**Comment by timlinux** [comment]
Hi @rischanlab 

I don't mind if we create an initial set of migrations from this point forward if that help - there are no other instances running out there that I know of.

Also can you get the test passing on travis?

Thanks!

Tim

**Comment by rischanlab** [comment]
Hi @timlinux How do you think about [this](https://github.com/pinax/django-user-accounts/issues/179) issue. I've tried to downgrade django version from 1.8.5 to 1.7.6 and it works, I can do migrate for `accounts userena` but still got error when I do migrate for `base` application. I' ve tried to migrate app one by one and I guess the problems are on `accounts` and `base` applications.

File: deployment/docker/REQUIREMENTS.txt [hunk 3]
```diff
@@ -26,3 +26,5 @@ django-disqus==0.5
 awesome-slugify==1.4
 #translation management tool
 django-rosetta==0.7.0
+#youtube embed
```
Comment (timlinux) [inline deployment/docker/REQUIREMENTS.txt]: lets keep this clean - remove if not needed

File: django_project/changes/templates/entry/detail.html [hunk 4]
```diff
@@ -9,62 +10,67 @@
...
+                <a href="{{ MEDIA_URL }}{{ entry.image_file }}">
+                    <img class="img-responsive img-rounded pull-right"
+                         src="{{ entry.image_file|thumbnail_url:'large-entry' }}"
+                         alt=""/>{# see core/settings/contrib.py for large-entry #}
+                </a>
+            {% endif %}
+        </div>
+        {% if entry.video %}
+            <div class="col-lg-8">
+                {% video entry.video 'small' %}
```
Comment (timlinux) [inline django_project/changes/templates/entry/detail.html]: can we center this in the div please (horizontally)

**Comment by rischanlab** [comment]
HI @timlinux could you check my PR update, I've center the video (see the screenshoot below). I've fixed the help text for input entry based on your comments. 

and for base_markdown, we need it, without that the page does not support markdown.
screen shot 2015-10-22 at 11 26 57 am

**Comment by timlinux** [comment]
f57f6556-78b0-11e5-9d89-e11c42242d56

@rischanlab can you put the credits above the comments please
@rischanlab and center the video in the left content area as shown. Might be good to make it bigger too so that it fills as much of that left space as possible....

**Comment by rischanlab** [comment]
Hi @timlinux it this LGTM I'll merge it. 
Sorry this PR is still from my develop branch, starting after this I'll make it new branch in each PR which is for one ticket. 

This is the screenshoot according to your comments 
* make a video center under the description column
* move the funder name above the comments. 

screen shot 2015-10-26 at 7 32 06 am

**Comment by timlinux** [comment]
Lgtm thanks

**Pull request merged**